int/api/massa/addresses.go

```diff
@@ -23,28 +23,36 @@ func AddressesHandler(params operations.MassaGetAddressesParams) middleware.Resp
 			)
 	}
 
-	//nolint: prealloc
-	var pendingBalances []string
-	for _, addressDetails := range addressesDetails {
-		//nolint: staticcheck, nolintlint
-		pendingBalances = append(pendingBalances, addressDetails.CandidateBalance)
-	}
-
-	//nolint: prealloc
-	var finalBalances []string
-	for _, addressDetails := range addressesDetails {
-		//nolint: staticcheck, nolintlint
-		finalBalances = append(finalBalances, addressDetails.CandidateBalance)
-	}
-
-	addressMap := make(models.AddressesAttributes, len(addressesDetails))
+	addressMap := make(map[string]operations.MassaGetAddressesOKBodyAddressesAttributesAnon, len(addressesDetails))
 
 	for _, details := range addressesDetails {
-		balance := &models.AddressesAttributesAnonBalance{Pending: details.CandidateBalance, Final: details.FinalBalance}
+		//nolint: exhaustruct
+		attribute := operations.MassaGetAddressesOKBodyAddressesAttributesAnon{}
+
+		if requestedAttributesContains(params, "balance") || requestedAttributeIsEmpty(params) {
+			attribute.Balance = &operations.MassaGetAddressesOKBodyAddressesAttributesAnonBalance{
+				Pending: details.CandidateBalance,
+				Final:   details.FinalBalance}
```

Comment:
should solve the linter problem according to my personal go linter :-)
```suggestion
				Final:   details.FinalBalance,
			}
```